ecs-agent/tcs/handler/handler_test.go

```diff
@@ -483,8 +570,73 @@ func TestStartTelemetrySessionMetricsChannelPauseWhenClientClosed(t *testing.T)
 
 	// simulating retry after backoff
 	newCtx, _ := context.WithCancel(context.Background())
-	go session.StartTelemetrySession(newCtx, server.URL)
+	go session.StartTelemetrySession(newCtx)
 	for len(telemetryMessages) != 0 {
 		time.Sleep(1 * time.Second)
 	} // test will time out if after tcs client startup, message does not resume flowing
 }
+
+// TestPeriodicDisconnectonTCSClient tests for tcs session disconnects at regular intervals.
+func TestPeriodicDisconnectonTCSClient(t *testing.T) {
+
+	// Start test server.
+	closeWS := make(chan []byte)
+	server, _, requestChan, serverErr, err := wsmock.GetMockServer(closeWS)
+	server.StartTLS()
+	defer server.Close()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	go func() {
+		for {
+			select {
+			case <-requestChan:
+			}
+		}
+	}()
+
+	testecsclient := &wsmock.TestECSClient{
+		TCSurl: server.URL,
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	deregisterInstanceEventStream := eventstream.NewEventStream("Deregister_Instance", ctx)
+	deregisterInstanceEventStream.StartListening()
+	defer cancel()
+
+	telemetryMessages := make(chan ecstcs.TelemetryMessage, testTelemetryChannelDefaultBufferSize)
+	healthMessages := make(chan ecstcs.HealthMessage, testTelemetryChannelDefaultBufferSize)
+
+	//Setting disconnect timer(10 secs) to be less than heartbeat timer( 1min)
+	//in order disconnect because of periodic timer instead of heartbeat timer due to inactivity.
```

Comment:
super nit since there are already other nit comments 🥲 - space after `//` and adjust parentheses